src/main/scala/regmapper/Annotation.scala

```diff
@@ -2,22 +2,188 @@
 
 package freechips.rocketchip.regmapper
 
-import org.json4s.JsonDSL._
-import org.json4s.jackson.JsonMethods.{pretty, render}
-
-object RegMappingAnnotation {
-  def serialize(base: BigInt, name: String, mapping: RegField.Map*): String = {
-    val regDescs = mapping.flatMap { case (byte, seq) =>
-      seq.map(_.width).scanLeft(0)(_ + _).zip(seq).map { case (bit, f) =>
-        val anonName = s"unnamedRegField${byte.toHexString}_${bit}"
-        (f.desc.map{ _.name}.getOrElse(anonName)) -> f.toJson(byte, bit)
-      }
+import chisel3.core.annotate
+import chisel3.experimental.{ChiselAnnotation, RawModule, RunFirrtlTransform}
+import firrtl.annotations._
+import firrtl.{CircuitForm, CircuitState, LowForm, Transform}
+
+case class RegFieldDescAnnotation(
+  target: ModuleName,
+  desc: String) extends SingleTargetAnnotation[ModuleName] {
+  def duplicate(n: ModuleName): RegFieldDescAnnotation = this.copy(n)
+}
+
+
+/** ********************************************************************************/
+
+case class RegFieldDescSer(
+  byteOffset: String,
+  bitOffset: Int,
+  bitWidth: Int,
+  name: String,
+  resetValue: BigInt,
+  accessType: String,
+  wrType: String,
+  rdAction: String,
+  desc: String,
+  group: String,
+  groupDesc: String,
+  volatile: Boolean = false,
+  hasReset: Boolean = false,
+  enumerations: Map[BigInt, (String, String)] = Map()
+)
+
+case class RegFieldSer(
+  regFieldName: String,
+  desc: RegFieldDescSer
+)
+
+case class RegistersSer(
```

Comment:
I don't think you need this extra level of case class, see below

Reply:
I am going to leave this for now.